test/integration/src/py/test_sifnode_peggy1_pause_unpause.py

```diff
@@ -0,0 +1,114 @@
+
+# We assert a tx is successful before pausing because we test the pause
+# functionality by 1. An error response and 2. Balance unchanged within timeout.
+# We want to make sure #2 is not a false positive due to lock function not
+# working in the first place
+def test_pause_lock_valid(ctx: EnvCtx):
+    # Test a working flow:
+    fund_amount_sif = 10 * test_utils.sifnode_funds_for_transfer_peggy1
+    fund_amount_eth = 1 * eth.ETH
+
+    test_sif_account = ctx.create_sifchain_addr(fund_amounts=[[fund_amount_sif, "rowan"]])
+    ctx.tx_bridge_bank_lock_eth(ctx.eth_faucet, test_sif_account, fund_amount_eth)
+    ctx.eth.advance_blocks()
+    # Setup is complete, test account has rowan AND eth
+
+    test_eth_destination_account = ctx.create_and_fund_eth_account()
+
+    send_amount = 1
+    balance_diff, erc_diff = send_test_account(ctx, test_sif_account, test_eth_destination_account, send_amount, erc20_token_addr=ctx.get_bridge_token_sc().address)
+    # TODO: sif_tx_fee vs get from envctx vs more lenient assertion
+    assert balance_diff.get(sifchain.ROWAN, 0) == (-1 * (send_amount + sifchain.sif_tx_fee_in_rowan )), "Gas fee and sent amount should be deducted from sender sif acct"
+    assert erc_diff == send_amount, "Eth destination should receive rowan token"
+
+    res = ctx.sifnode.pause_peggy_bridge(ctx.sifchain_ethbridge_admin_account)
+    assert res[0]['code'] == 0
+
+    balance_diff, erc_diff = send_test_account(ctx, test_sif_account, test_eth_destination_account, send_amount, erc20_token_addr=ctx.get_bridge_token_sc().address)
+    assert balance_diff.get(sifchain.ROWAN, 0) == (-1 * sifchain.sif_tx_fee_in_rowan), "Only gas fee should be deducted for attempted tx"
+    assert erc_diff == 0, "Eth destination should not receive rowan token"
+
+    res = ctx.sifnode.unpause_peggy_bridge(ctx.sifchain_ethbridge_admin_account)
+    assert res[0]['code'] == 0
+    # # Submit lock
+    # # Assert tx go through, balance updated correctly.
+    send_amount = 15
+    balance_diff, erc_diff = send_test_account(ctx, test_sif_account, test_eth_destination_account, send_amount, erc20_token_addr=ctx.get_bridge_token_sc().address)
+    # TODO: sif_tx_fee vs get from envctx vs more lenient assertion
+    assert balance_diff.get(sifchain.ROWAN, 0) == (-1 * (send_amount + sifchain.sif_tx_fee_in_rowan )), "Gas fee and sent amount should be deducted from sender sif acct"
+    assert erc_diff == send_amount, "Eth destination should receive rowan token"
+
+# Burn CETH
+def test_pause_burn_valid(ctx: EnvCtx):
+    fund_amount_sif = 10 * test_utils.sifnode_funds_for_transfer_peggy1
+    fund_amount_eth = 1 * eth.ETH
+
+    test_sif_account = ctx.create_sifchain_addr(fund_amounts=[[fund_amount_sif, "rowan"]])
+    ctx.tx_bridge_bank_lock_eth(ctx.eth_faucet, test_sif_account, fund_amount_eth)
+    ctx.eth.advance_blocks(100)
+    # Setup is complete, test account has rowan AND eth
+    # Sleep for relayer to complete transaction
+    time.sleep(5)
```

Comment:
Do we not have a way to wait on balance change with timeout rather then just sleeping?